packages/ra-ui-materialui/src/input/ArrayInput/SimpleFormIteratorItem.tsx

```diff
@@ -77,6 +80,25 @@ export const SimpleFormIteratorItem = React.forwardRef(
                 ? getItemLabel(index)
                 : getItemLabel;
 
+        const parentSourceContext = useSourceContext();
+        const sourceContext = useMemo(
+            () => ({
+                getSource: (source: string) =>
+                    source ? `${member}.${source}` : member,
+                getLabel: (source: string) => {
+                    // remove digits, e.g. 'book.authors.2.categories.3.identifier.name' => 'book.authors.categories.identifier.name'
+                    const itemSource = `${member.replace(
+                        /\.\d+/g,
+                        ''
+                    )}.${source}`;
```

Comment:
Not sure you need this, since, if I'm not mistaken, `getLabel` will be called with the original `source` (i.e. `"name"`) and not the `finalSource` (i.e. `'book.authors.2.categories.3.identifier.name'`)

Reply:
Nope because for instance with an `<ArrayInput source="authors">` containing a `<TextInput source="name">`  we want the label key to `authors.name` not `name`

Reply:
Okay I get it, I missed the fact that you were applying the `replace` on `member` and not `source`.

However this is still missing something to me: just like for `getSource`, you need to check if `source` is defined before concatenating with `.${source}`.